src/components/comments/CommentBox.jsx

```diff
@@ -1,215 +1,168 @@
-import React from "react";
-import Badge from "react-bootstrap/Badge";
-import axios from "axios";
-import moment from "moment";
-import CommentList from "./CommentList";
-import CommentForm from "./CommentForm";
-import { LOCAL_HOST } from "../../consts";
-import DropdownButton from "react-bootstrap/DropdownButton";
-import DropdownMenu from "react-bootstrap/DropdownMenu";
+import React, { useContext, useEffect, useState } from "react";
 import Dropdown from "react-bootstrap/Dropdown";
-import { v4 as uuidv4 } from 'uuid';
+import { CommentContext } from "../../contexts/comment-context";
+import CommentForm from './CommentForm';
+import CommentList from './CommentList';
 
-class CommentBox extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { comments: [], votedComments: {}, userID: "", userName: "" };
-
-    this.getComments = this.getComments.bind(this);
-    this.handleCommentSubmit = this.handleCommentSubmit.bind(this);
-    this.createAnonymousUser = this.createAnonymousUser.bind(this);
-    this.getVotesForUser = this.getVotesForUser.bind(this);
+import { feedbackAPI, userAPI } from "../../api";
 
+const CommentBox = (props) => {
+  const [state, dispatch] = useContext(CommentContext);
+  const [sortButtonText, setSortButtonText] = useState("Sort by Time");
+  // get feedbacks
+  useEffect(() => {
     // check if there is a user credential stored in localStorage
     // if not, create a new user credential
-    let storedUserStr = localStorage.getItem('user');
+    const storedUserStr = localStorage.getItem('user');
+    let userID = "";
+    let userName = "";
     if (storedUserStr === null) {
-      this.createAnonymousUser();
+      createAnonymousUser();
```

Comment:
Also, could we move this logic of creating user to `App.js` or somewhere global? I need `userId` even for building of projects.